core/geometry/src/test/polyface/PolyfaceQuery.test.ts

```diff
@@ -1415,3 +1425,265 @@ describe("FacetShape", () => {
+        PolyfaceClip.clipPolyfaceUnionOfConvexClipPlaneSetsToBuilders(mesh0, clipper, builders, 1);
+        mesh1 = builders.claimPolyface(1, true);
+        GeometryCoreTestIO.captureCloneGeometry(allGeometry, mesh1, x0 += 2, 0, z0);
+      }
+    }
+
+    // compare output of sweepLineStringToFacets with and without searcher
+    if (ck.testDefined(contour, "created hole contour")) {
+      let drapePath: IndexedXYZCollection | undefined;
+      contour.computeXYStrokes();
+      const toDrape = contour.xyStrokes;
+      if (ck.testDefined(toDrape, "stroked the contour"))
+        if (ck.testType(toDrape, Loop, "strokes are contained in a Loop"))
+          if (ck.testExactNumber(1, toDrape.children.length, "Loop has 1 child"))
+            if (ck.testType(toDrape.children[0], LineString3d, "child is a LineString3d")) {
+              const worldPoints = toDrape.children[0].packedPoints.clone();
+              worldPoints.multiplyTransformInPlace(contour.localToWorld);
+              drapePath = worldPoints;
+            }
+      if (ck.testDefined(drapePath, "ready to drape")) {
+        timer.start();
+        const sweptEdges0 = PolyfaceQuery.sweepLineStringToFacets(drapePath, mesh0);
+        const elapsedDrape = timer.stop().milliseconds;
+
+        const sweepOptions = SweepLineStringToFacetsOptions.create();
+        const xyRange = Range2d.createFrom(mesh0.range());
+        const searcher = GriddedRaggedRange2dSetWithOverflow.create<number>(xyRange, subdivisions, subdivisions)!;
+        for (const visitor = mesh0.createVisitor(0); visitor.moveToNextFacet();)
+          searcher.addRange(visitor.point.getRange(), visitor.currentReadIndex());
+        sweepOptions.searcher = searcher;
+
+        timer.start();
+        const sweptEdges1 = PolyfaceQuery.sweepLineStringToFacets(drapePath, mesh0, sweepOptions);
+        const elapsedDrapeWithSearcher = timer.stop().milliseconds;
+
+        timer.start();
+        const sweptEdges2 = PolyfaceQuery.sweepLineStringToFacetsXY(drapePath, mesh0, sweepOptions.searcher);
+        const elapsedDrapeXY = timer.stop().milliseconds;
+
+        ck.testLE(elapsedDrapeWithSearcher, elapsedDrape, "searcher speeds up drape");
+        ck.testLE(elapsedDrapeXY, elapsedDrape, "drapeXY speeds up vertical drape");
+
+        const hole0 = BagOfCurves.create(...sweptEdges0).collectCurvePrimitives(undefined, true, true) as LineSegment3d[];
+        const hole1 = BagOfCurves.create(...sweptEdges1).collectCurvePrimitives(undefined, true, true) as LineSegment3d[];
+        const hole2 = BagOfCurves.create(...sweptEdges2).collectCurvePrimitives(undefined, true, true) as LineSegment3d[];
+        ck.testTrue(compareLineSegmentArrays(hole0, hole1, true), "drape compares to drape with searcher");
+        ck.testTrue(compareLineSegmentArrays(hole1, hole2, true), "drape with searcher compares to drapeXY");
+
+        GeometryCoreTestIO.captureCloneGeometry(allGeometry, hole0, x0, 0, z0 += 2);
```

Comment:
any reason for not capturing `hole2`?

Reply:
Yes, because `hole0` and `hole2` represent existing logic with equivalent output. And I already capture `hole0`. I am just testing here that the output of the accelerated branch of `sweepLineStringToFacets` agrees with both its non-accelerated branch, and with the output of `sweepLineStringToFacetsXY`.